cli/src/main/java/bisq/cli/app/BisqCliMain.java

```diff
@@ -41,60 +45,28 @@
 
     private final ManagedChannel channel;
     private final CliCommand cmd;
+    private final OptionParser cmdParser;
 
     public static void main(String[] args) {
-        new BisqCliMain("localhost", 8888);
+        new BisqCliMain("localhost", 8888, args);
     }
 
-    private BisqCliMain(String host, int port) {
+    private BisqCliMain(String host, int port, String[] params) {
         // Channels are secure by default (via SSL/TLS);  for the example disable TLS to avoid needing certificates.
         this(ManagedChannelBuilder.forAddress(host, port).usePlaintext().build());
 
-        // Simple input scanner
-        // TODO use some more sophisticated input processing with validation....
-        try (Scanner scanner = new Scanner(in)) {
-            while (true) {
-                long startTs = currentTimeMillis();
-
-                String[] tokens = scanner.nextLine().split(" ");
-                if (tokens.length == 0) {
-                    return;
-                }
-                String command = tokens[0];
-                if (tokens.length > 1) {
-                    List<String> params = new ArrayList<>(Arrays.asList(tokens));
-                    params.remove(0);
-                }
-                String result;
-
-                switch (command) {
-                    case "getBalance":
-                        long satoshis = cmd.getBalance();
-                        // TODO mimic bitcoin-cli?  Depends on an error code: Loading block index... Verifying blocks...
-                        result = satoshis == -1 ? "Server initializing..." : cmd.prettyBalance.apply(satoshis);
-                        break;
-                    case "getVersion":
-                        result = cmd.getVersion();
-                        break;
-                    case "stop":
-                        result = "Shut down client";
-                        try {
-                            shutdown();
-                        } catch (InterruptedException e) {
-                            log.error(e.toString(), e);
-                        }
-                        break;
-                    case "stopServer":
-                        cmd.stopServer();
-                        result = "Server stopped";
-                        break;
-                    default:
-                        result = format("Unknown command '%s'", command);
-                }
-
-                // First response is rather slow (300 ms) but following responses are fast (3-5 ms).
-                log.info("{}\t{}", result, cmd.responseTime.apply(startTs));
-            }
+        long startTs = currentTimeMillis();
+
+        String command = parseCommand(params);
+        String result = runCommand(command);
+
+        // First response is rather slow (300 ms) but following responses are fast (3-5 ms).
+        // log.info("{}\t{}", result, cmd.responseTime.apply(startTs));
+        out.println(result + "\t" + cmd.responseTime.apply(startTs));
```

Comment:
And this as well. There is no requirement at present to measure the response time, and output from a unix utility should not include anything other than the data requested, i.e. such that it can be piped to the next utility. This sort of output breaks that unix idiom. Again, I realize you did not introduce this. I'm just asking you to remove it and start thinking about things this way, making similar decisions as you go.